src/components/flame-graph/index.js

```diff
@@ -54,38 +76,51 @@ class FlameGraph extends React.PureComponent<Props> {
       selection,
       threadName,
       processDetails,
+      selectedCallNodeIndex,
+      profileCallTreeContextMenuVisible,
     } = this.props;
 
     const maxViewportHeight = maxStackDepth * STACK_FRAME_HEIGHT;
 
     return (
       <div className="flameGraph">
         <FlameGraphSettings />
+        <TransformNavigator />
         <div className="flameGraphContent">
           <div title={processDetails} className="flameGraphLabels grippy">
             <span>{threadName}</span>
           </div>
-          <FlameGraphCanvas
-            key={threadIndex}
-            // ChartViewport props
-            viewportProps={{
-              timeRange,
-              maxViewportHeight,
-              maximumZoom: 1,
-              selection,
-              startsAtBottom: true,
-              disableHorizontalMovement: true,
-              viewportNeedsUpdate,
+          <ContextMenuTrigger
+            id={'ProfileCallTreeContextMenu'}
```

Comment:
Let's move the context menu into a `src/components/shared`. We could use it in the Stack Chart as well now!